connectivity/tests/lrp.go

```diff
@@ -188,3 +188,42 @@ func WaitForLocalRedirectBPFEntries(ctx context.Context, t *check.Test, frontend
 		return
 	}
 }
+
+// LRPWithNodeDNS runs test scenarios for local redirect policy
+// with the node local DNS setup.
+//
+// It sends HTTP requests to the ExternalTarget to check the DNS
+// requests are resolved by node-local DNS cache pods.
+// The network policy allows the clients to access node-local-dns
+// and the ExternalTarget.
+func LRPWithNodeDNS(opts ...RetryOption) check.Scenario {
+	cond := &retryCondition{}
+	for _, op := range opts {
+		op(cond)
+	}
+	return lrpWithNodeDNS{rc: cond}
+}
+
+type lrpWithNodeDNS struct {
+	rc *retryCondition
+}
+
+func (s lrpWithNodeDNS) Name() string {
+	return "local-redirect-policy-with-node-dns"
+}
+
+func (s lrpWithNodeDNS) Run(ctx context.Context, t *check.Test) {
+	ct := t.Context()
+
+	extTarget := t.Context().Params().ExternalTarget
```

Comment:
We should avoid external target, they are flaky. Let's use an in-cluster test service instead. Node-local DNS is typically used to improve performance for resolving in-cluster service FQDNs - https://kubernetes.io/docs/tasks/administer-cluster/nodelocaldns/#introduction.